bool is_next
---
bool success